cw_evm/lib/evm_chain_wallet.dart

```diff
@@ -0,0 +1,584 @@
+        'private_key': privateKey,
+        'balance': balance[currency]!.toJSON(),
+      });
+
+  static Future<EVMChainWallet> open({
+    required String name,
+    required String password,
+    required WalletInfo walletInfo,
+  }) async {
+    final path = await pathForWallet(name: name, type: walletInfo.type);
+    final jsonSource = await read(path: path, password: password);
+    final data = json.decode(jsonSource) as Map;
+    final mnemonic = data['mnemonic'] as String?;
+    final privateKey = data['private_key'] as String?;
+    final balance = EVMChainERC20Balance.fromJSON(data['balance'] as String) ??
+        EVMChainERC20Balance(BigInt.zero);
+
+    final nativeCurrency = currencyForWalletType(walletInfo.type);
+
+    return EVMChainWallet(
+      walletInfo: walletInfo,
+      password: password,
+      mnemonic: mnemonic,
+      privateKey: privateKey,
+      initialBalance: balance,
+      nativeCurrency: nativeCurrency,
+    );
+  }
+
+  Future<void> _updateBalance() async {
+    balance[currency] = await _fetchEVMChainBalance();
+
+    await _fetchErc20Balances();
+    await save();
+  }
+
+  Future<EVMChainERC20Balance> _fetchEVMChainBalance() async {
+    final balance = await _client.getBalance(_evmChainPrivateKey.address);
+    return EVMChainERC20Balance(balance.getInWei);
+  }
+
+  Future<void> _fetchErc20Balances() async {
+    for (var token in evmChainErc20TokensBox.values) {
+      try {
+        if (token.enabled) {
+          balance[token] = await _client.fetchERC20Balances(
+            _evmChainPrivateKey.address,
+            token.contractAddress,
+          );
+        } else {
+          balance.remove(token);
+        }
+      } catch (_) {}
+    }
+  }
+
+  Future<EthPrivateKey> getPrivateKey(
+      {String? mnemonic, String? privateKey, required String password}) async {
+    assert(mnemonic != null || privateKey != null);
+
+    if (privateKey != null) {
+      return EthPrivateKey.fromHex(privateKey);
+    }
+
+    final seed = bip39.mnemonicToSeed(mnemonic!);
+
+    final root = bip32.BIP32.fromSeed(seed);
+
+    const hdPathEVMChain = "m/44'/60'/0'/0";
+    const index = 0;
+    final addressAtIndex = root.derivePath("$hdPathEVMChain/$index");
+
+    return EthPrivateKey.fromHex(HEX.encode(addressAtIndex.privateKey as List<int>));
+  }
+
+  Future<void>? updateBalance() async => await _updateBalance();
+
+  List<Erc20Token> get erc20Currencies => evmChainErc20TokensBox.values.toList();
+
+  Future<void> addErc20Token(Erc20Token token) async {
+    String? iconPath;
+    try {
+      iconPath = CryptoCurrency.all
+          .firstWhere((element) => element.title.toUpperCase() == token.symbol.toUpperCase())
+          .iconPath;
+    } catch (_) {}
+
+    final _token = Erc20Token(
+      name: token.name,
+      symbol: token.symbol,
+      contractAddress: token.contractAddress,
+      decimal: token.decimal,
+      enabled: token.enabled,
+      tag: token.tag ?? "ETH",
```

Comment:
should not be the case for Polygon

Reply:
Yeah, adjusted it. Polygon previously had an override method for this.